packages/next/src/server/lib/router-server.ts

```diff
@@ -667,6 +667,13 @@ export async function initialize(opts: {
 
         // assetPrefix overrides basePath for HMR path
         if (assetPrefix) {
+          // if assetPrefix is a URL, it can break HMR
+          if (assetPrefix.startsWith('http')) {
+            throw new Error(
+              `assetPrefix is a URL (${assetPrefix}), which is production-only. For more information, see https://nextjs.org/docs/app/api-reference/next-config-js/assetPrefix`
```

Comment:
We should error earlier e.g. during config resolution phase. At this point, `assetPrefix` should not be a URL I guess. Mostly since it's not clear if we might invoke this function multiple times and then spam the console.

 The error should be explicit about `assetPrefix` not being supported during `next dev` which is not the same as saying "production-only" now that we support development `next build`.